### PR TITLE
[Backport release-24.05] voms: 2.1.0-rc2-unstable-2022-06-14 -> 2.1.0

### PR DESCRIPTION
**Pull request by ShamrockLee**
## Description of changes

This pull request backports #270400 and bumps `voms` onto the latest official and stable release, 2.1.0. To do that, I backported its dependent commit 17ce994aa460614ec8033676ea85670eaa45df0e, normalizing `name` and `version` for non-officially released revision.

This pull request also backports #325116. It formats the code according to Nix RFC 166 and moves it under `pkg/by-name` to ease maintenance and future backports.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed; also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage) (It rebuilds some large packages such as `root` (CERN ROOT) and takes some time to run.)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`) (`voms --help` and `voms --version` works. `voms-proxy-init` works and creates usable X509 user proxy on CERN LXPLUS.)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ShamrockLee** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.gfal2-python</li>
    <li>python312Packages.gfal2-python.dist</li>
    <li>python312Packages.gfal2-util</li>
    <li>python312Packages.gfal2-util.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>gfal2</li>
    <li>gfal2-util (python311Packages.gfal2-util)</li>
    <li>gfal2-util.dist (python311Packages.gfal2-util.dist)</li>
    <li>hepmc3</li>
    <li>htcondor</li>
    <li>ns-3</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
    <li>python311Packages.gfal2-python</li>
    <li>python311Packages.gfal2-python.dist</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.snakemake-storage-plugin-xrootd</li>
    <li>python311Packages.snakemake-storage-plugin-xrootd.dist</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.coffea</li>
    <li>python312Packages.coffea.dist</li>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.snakemake-storage-plugin-xrootd</li>
    <li>python312Packages.snakemake-storage-plugin-xrootd.dist</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>voms</li>
    <li>voms.bin</li>
    <li>voms.dev</li>
    <li>voms.etc</li>
    <li>voms.man</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by ShamrockLee** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.gfal2-python</li>
    <li>python312Packages.gfal2-python.dist</li>
    <li>python312Packages.gfal2-util</li>
    <li>python312Packages.gfal2-util.dist</li>
  </ul>
</details>
<details>
  <summary>40 packages built:</summary>
  <ul>
    <li>gfal2</li>
    <li>gfal2-util (python311Packages.gfal2-util)</li>
    <li>gfal2-util.dist (python311Packages.gfal2-util.dist)</li>
    <li>hepmc3</li>
    <li>htcondor</li>
    <li>ns-3</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
    <li>python311Packages.gfal2-python</li>
    <li>python311Packages.gfal2-python.dist</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.snakemake-storage-plugin-xrootd</li>
    <li>python311Packages.snakemake-storage-plugin-xrootd.dist</li>
    <li>python311Packages.xrootd</li>
    <li>python311Packages.xrootd.dist</li>
    <li>python312Packages.coffea</li>
    <li>python312Packages.coffea.dist</li>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
    <li>python312Packages.hepmc3</li>
    <li>python312Packages.snakemake-storage-plugin-xrootd</li>
    <li>python312Packages.snakemake-storage-plugin-xrootd.dist</li>
    <li>python312Packages.xrootd</li>
    <li>python312Packages.xrootd.dist</li>
    <li>root</li>
    <li>root.debug</li>
    <li>voms</li>
    <li>voms.bin</li>
    <li>voms.dev</li>
    <li>voms.etc</li>
    <li>voms.man</li>
    <li>xfitter</li>
    <li>xrootd</li>
    <li>xrootd.bin</li>
    <li>xrootd.dev</li>
    <li>xrootd.etc</li>
    <li>xrootd.man</li>
    <li>yoda-with-root</li>
  </ul>
</details>

File: pkgs/by-name/vo/voms/package.nix [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
   # create a new output "etc", move "$out/etc" to "$etc/etc"
   # and symlink "$out/etc" to externalEtc.
-, externalEtc ? "/etc"
+  externalEtc ? "/etc",
 }:
 
-stdenv.mkDerivation rec{
-  pname = "voms-unstable";
-  version = "2022-06-14";
+stdenv.mkDerivation (finalAttrs: {
```
Comment (superherointj) [inline pkgs/by-name/vo/voms/package.nix:22]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```
FYI `rec` may be faster on eval than a [fixed point](https://github.com/NixOS/nixpkgs/issues/320528#issuecomment-2175087036). Changing this for the sake of changing it I don't know if it is a good idea.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 325097` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Review by superherointj — APPROVED** [review]

File: pkgs/by-name/vo/voms/package.nix (hunk 1 above)
Comment (ShamrockLee) [reply pkgs/by-name/vo/voms/package.nix:22]: > FYI rec may be faster on eval than a fixed point. Changing this for the sake of changing it I don't know if it is a good idea.

Glad to hear about the cons of the fixed-point arguments. Thank you for this suggestions.

**Pull request merged**